components/textfield/stories/textfield.stories.js

```diff
@@ -122,24 +141,133 @@ export default {
+*/
+export const Disabled = Template.bind({});
+Disabled.tags = ["!dev"];
+Disabled.args = {
+	isDisabled: true
 };
-TextArea.parameters = {
-	chromatic: { disableSnapshot: true },
+Disabled.parameters = {
+	chromatic: { disableSnapshot: true }
 };
 
+
+/**
+ * A text field can be marked as having an error to show that a value needs to be entered in order to move forward or that a value that was entered is invalid. If an error exists, the error icon always overrides the validation icon.
+*/
+export const Invalid = Template.bind({});
+Invalid.tags = ["!dev"];
+Invalid.args = {
+	isInvalid: true
+};
+Invalid.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+
+/**
+ * A text field can have [help text](/docs/components-help-text--docs) below the field to give extra context or instruction about what a user should input in the field. The help text area has two options: a description and an error message. The description communicates a hint or helpful information, such as specific requirements for correctly filling out the field. The error message communicates an error for when the field requirements aren’t met, prompting a user to adjust what they had originally input.
+ *
+ * Instead of placeholder text, use the help text description to convey requirements or to show any formatting examples that would help user comprehension. Putting instructions for how to complete an input, requirements, or any other essential information into placeholder text is not accessible.
+*/
+export const HelpText = HelpTextOptions.bind({});
+HelpText.tags = ["!dev"];
+HelpText.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+
+
+export const Quiet = QuietGroup.bind({});
+Quiet.tags = ["!dev"];
+Quiet.args = {
+	isQuiet: true,
+	value: ""
+};
+Quiet.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+
+/**
+ * Text fields have a read-only option for when content in the disabled state still needs to be shown. This allows for content to be copied, but not interacted with or changed.
+*/
+export const Readonly = Template.bind({});
+Readonly.tags = ["!dev"];
+Readonly.args = {
+	isReadOnly: true,
+	value: "lisawilson24"
+};
+Readonly.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+Readonly.storyName = "Read-only";
+
+/**
+ * Side labels are most useful when vertical space is limited.
+*/
+export const SideLabel = Template.bind({});
+SideLabel.tags = ["!dev"];
+SideLabel.args = {
+	labelPosition: "side",
+	hasCharacterCount: true,
+	characterCount: 50,
+	helpText: "Example help text. Lorem ipsum dolor sit amet."
+};
+SideLabel.parameters = {
+	chromatic: { disableSnapshot: true }
+};
+
+export const Sizing = (args, context) => Sizes({
+	Template: Template,
+	withHeading: false,
+	withBorder: false,
+	...args,
+}, context);
+Sizing.args = {
+	helpText: "Example help text. Lorem ipsum dolor sit amet."
```

Comment:
Could we add `hasCharacterCount: true` here? So that people can see character count also has its own size changes? 

<img width="294" alt="Screenshot 2024-10-18 at 10 04 41 AM" src="https://github.com/user-attachments/assets/c727b302-fb6e-4974-819f-60b344d497d2">